community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -208,6 +211,25 @@ default void close() throws TransactionFailureException
 
     Revertable restrict( AccessMode mode );
 
+    /**
+     * @return list of executing queries that use this transaction
+     */
+    Stream<ExecutingQuery> executingQueries();
```

Comment:
Can multiple queries really share a tx simultaneously? I would have thought that this was an `Option<ExecutingQuery>`